fontra2fontir/src/toir.rs

```diff
@@ -0,0 +1,298 @@
+        if first_type.is_off_curve() {
+            return Err(WorkError::InvalidSourceGlyph {
+                glyph_name: glyph_name.clone(),
+                message: String::from("Open path starts with off-curve points"),
+            });
+        }
+        path_builder.move_to((first.x, first.y))?;
+        add_to_path(
+            glyph_name.clone(),
+            &mut path_builder,
+            contour.points[1..].iter(),
+        )?;
+    } else {
+        add_to_path(glyph_name.clone(), &mut path_builder, contour.points.iter())?;
+    }
+
+    let path = path_builder.build()?;
+    trace!(
+        "Built a {} entry path for {}",
+        path.elements().len(),
+        glyph_name
+    );
+    Ok(path)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+
+    use fontdrasil::{coords::NormalizedCoord, types::Axis};
+    use fontir::ir::Glyph;
+    use kurbo::{BezPath, PathEl};
+    use write_fonts::types::Tag;
+
+    use crate::{
+        fontra::{FontraFontData, FontraGlyph},
+        test::testdata_dir,
+        toir::to_ir_static_metadata,
+    };
+
+    use super::to_ir_glyph;
+
+    fn axis_tuples(axes: &[Axis]) -> Vec<(&str, Tag, f64, f64, f64)> {
+        axes.iter()
+            .map(|a| {
+                (
+                    a.name.as_str(),
+                    a.tag,
+                    a.min.to_f32() as f64,
+                    a.default.to_f32() as f64,
+                    a.max.to_f32() as f64,
+                )
+            })
+            .collect::<Vec<_>>()
+    }
+
+    fn commands(b: &BezPath) -> String {
+        b.elements()
+            .iter()
+            .map(|e| match e {
+                PathEl::MoveTo(..) => "M",
+                PathEl::LineTo(..) => "L",
+                PathEl::QuadTo(..) => "Q",
+                PathEl::CurveTo(..) => "C",
+                PathEl::ClosePath => "Z",
+            })
+            .collect::<Vec<_>>()
+            .join("")
+    }
+
+    fn assert_contour_compatibility(glyph: &Glyph) {
+        // compatible if all sources have the same drawing commands in the same order
+        let unique_command_seqs = glyph
+            .sources()
+            .iter()
+            .map(|(_, s)| {
+                s.contours
+                    .iter()
+                    .map(commands)
+                    .collect::<Vec<_>>()
+                    .join(" ")
+            })
+            .collect::<HashSet<_>>();
+        assert_eq!(1, unique_command_seqs.len(), "{unique_command_seqs:?}");
+    }
+
+    #[test]
+    fn static_metadata_of_2glyphs() {
+        let fontdata_file = testdata_dir().join("2glyphs.fontra/font-data.json");
+        let font_data = FontraFontData::from_file(&fontdata_file).unwrap();
+        let static_metadata = to_ir_static_metadata(&font_data).unwrap();
+        assert_eq!(1000, static_metadata.units_per_em);
+        assert_eq!(
+            vec![
+                ("Weight", Tag::from_be_bytes(*b"wght"), 200.0, 200.0, 900.0),
```

Comment:
`Tag::new(b"wght")` is a bit more concise and doesn't require the dereference.

Reply:
Oops, will fix